lib/packages/util/src/burstValue.ts

```diff
@@ -0,0 +1,177 @@
+/** @module util */
+
+/**
+ * Original work Copyright (c) 2019 Burst Apps Team
+ */
+
+import Big from 'big.js';
+
+Big.NE = -9;
+
+export enum BurstValueFormat {
+    PLANCK,
+    BURST,
+}
+
+/**
+ * A Value Object to facilitate BURST and Planck conversions/calculations.
+ *
+ * Note: This class uses a big number representation (ES5 compatible) under the hood, so
+ * number limits and numeric calculations are much more precise than JS number type
+ */
+export class BurstValue {
+
+    private _planck: Big;
+
+    private constructor(planck: string) {
+        this._planck = Big(planck);
+    }
+
+    /**
+     * Creates a Burst Value object from Planck
+     * @param planck The value in Planck
+     */
+    public static fromPlanck(planck: string) {
+        return new BurstValue(planck);
+    }
+
+    /**
+     * Creates a Burst Value object from BURST
+     * @param burst The value in BURST
+     */
+    public static fromBurst(burst: number|string) {
+        const b = new BurstValue('0');
+        b.setBurst(typeof burst === 'number' ? burst.toString(10) : burst);
+        return b;
+    }
+
+    /**
+     * Leaky value getter
+     * @return the underlying value in its big number representation (immutable)
+     */
+    getRaw(): Big {
+        return Big(this._planck);
+    }
+
+    /**
+     * @return Gets Planck representation
+     */
+    getPlanck(): string {
+        return this._planck.toString();
```

Comment:
Any reason not to  use getters and setters? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/set

Reply:
I did it initially, but the angular compiler complained, as getters and setters are not supported by TS <3.6 _and_ the wallet does not build wth TS >=3.6. So, before updating ng to be buildable by newer TS compiler I did it the "Java" way. I don't like it either....I was even thinking of using a Proxy, but I opted for the most straightforward way.